src/app/(main)/dashboard/components/voyage-dashboard/VoyageDashboard.tsx

```diff
@@ -8,20 +6,51 @@ import IdeationStateContent from "./IdeationStateContent";
 import FeaturesStateContent from "./FeaturesStateContent";
 import TechStackStateContent from "./TechStackStateContent";
 import ResourcesStateContent from "./ResourcesStateContent";
-import useVoyageDashboardLogic from "./useVoyageDashboardLogic";
-import { CHECKIN_STATUS } from "@/app/(main)/dashboard/mocks/voyageDashboardData";
+import { EventList, getDashboardData } from "./getDashboardData";
+import {
+  CHECKIN_STATUS,
+  getFeaturesData,
+  getIdeationData,
+  getResourcesData,
+  getTechStackData,
+} from "@/app/(main)/dashboard/mocks/voyageDashboardData";
 import VoyageSupport from "@/app/(main)/dashboard/components/shared/VoyageSupport";
-import { useAuth } from "@/store/hooks";
+import EmptySprintProvider from "@/app/(main)/my-voyage/[teamId]/sprints/providers/EmptySprintProvider";
+import { getUser } from "@/utils/getUser";
+import { Sprint } from "@/store/features/sprint/sprintSlice";
+
+interface VoyageDashboardProps {
+  teamId?: string;
+}
+async function VoyageDashboard({ teamId }: VoyageDashboardProps) {
+  //NOTE - Mocked value to show the filled state dashboard
+  const filledState = true;
+  const ideationData = filledState ? getIdeationData() : null;
+  const featureData = filledState ? getFeaturesData() : null;
+  const techStackData = filledState ? getTechStackData() : null;
+  const resourceData = filledState ? getResourcesData() : null;
+
+  const [user, error] = await getUser();
+
+  let currentSprintNumber: number | null = null;
+  let sprintsData: Sprint[] = [];
+  let meetingsData: EventList[] = [];
 
-function VoyageDashboard() {
-  //NOTE - This is a custom hook that returns mock data based on the filledState
-  const { ideationData, featureData, techStackData, resourceData, sprintData } =
-    useVoyageDashboardLogic(true);
+  if (teamId !== undefined) {
+    const data = await getDashboardData(user, error, Number(teamId));
+    currentSprintNumber = data.currentSprintNumber;
+    sprintsData = data.sprintsData;
+    meetingsData = data.meetingsData;
+  }
 
-  return (
-    <div className="grid grid-cols-2 gap-x-6 w-full">
+  return user ? (
+    <div className="flex flex-col min-[1470px]:grid min-[1470px]:grid-cols-2 gap-x-6 max-[1470px]:gap-y-6 w-full">
       <div className="col-span-1 flex flex-col gap-y-6 flex-grow-2">
-        <CalendarWidget sprintData={sprintData ?? undefined} />
```

Comment:
After implementing the requested changes, this ternary with the user won't be necessary. Also in the prevoyage dashboard page, you don't need to do a wrapper and check for the auth status there either.